js/src/index.spec.ts

```diff
@@ -185,13 +185,15 @@ describe("Schematic", () => {
 
   describe("initialize", () => {
     beforeAll(() => {
-      Object.defineProperty(window, "onbeforeunload", {
-        value: null,
-        writable: true,
-      });
+      if (typeof window !== "undefined") {
+        Object.defineProperty(window, "onbeforeunload", {
+          value: null,
+          writable: true,
+        });
+      }
     });
 
-    it("should flush event queue on beforeunload event", () => {
+    (typeof window === "undefined" ? it.skip : it)("should flush event queue on beforeunload event", () => {
```

Comment:
this event queue is used when waiting for the document to be ready; it's not used in react-native, so this test is not relevant